cmake/cpu_features.cmake

```diff
@@ -60,6 +60,23 @@ elseif (ARCH_AARCH64)
         set (COMPILER_FLAGS "${COMPILER_FLAGS} -march=armv8.2-a+simd+crypto+dotprod+ssbs -Xclang=-target-feature -Xclang=+ldapr -Wno-unused-command-line-argument")
     endif ()
 
+    # Best-effort check: The build generates and executes intermediate binaries, e.g. protoc and llvm-tablegen. If we build on ARM for ARM
```

Comment:
Why don't we simply build the host tools without all the extra instruction sets?

Reply:
New toolchains unfortunately won't help because the problem is not because of the host tools, it is because of generated intermediate tools `protoc` and `llvm-tablegen`.

Reply:
I name what we build to use on the host machine "host tools" (as we build for two platforms: "host" and "target").
So, the question is - why don't always build them without all the extra instruction sets? (or even -march=native will suffice).

Reply:
I see, it is a valid point. Let me have a brief look, shouldn't be too difficult to achieve that.

Reply:
I checked. Building host tools differently (e.g. with -march=native) is possible but ugly. E.g. `llvm-tblgen` links a ton of LLVM libs for which we also need to change `CMAKE_CXX_FLAGS` and these build descriptions are scattered all over the place in contrib/llvm-project/. Even more ugly, the same libraries are linked into ClickHouse, which means we would need to build them twice for modern and for basic CPU flags. `protoc` more or less has the same problem. Perhaps this will become simpler when llvm and protobuf are bundled.

Since only few users are affected, I would prefer to keep it as is for now.